### PR TITLE
Fix warnings/errors on MSVC

### PR DESCRIPTION
**Pull request by soarqin**
1. update googletest to tag release-1.11.0 to make googletest and googlemock compile on MSVC2022.
2. set gtest_force_shared_crt to ON in test cmake project to make tests compile on MSVC.
3. add /wd4251 and /wd4275 to MSVC compile flags to ignore DLL related warnings for class exports.
4. fix some other warnings and errors while compiling on MSVC2022.

### CODE REVIEW

File: test/mock/mock_IGameCallback.h [hunk 1]
```diff
@@ -23,10 +23,10 @@ class GameCallbackMock : public IGameCallback
 public:
 	using UpperCallback = ::ServerCallback;
 
-	GameCallbackMock(UpperCallback * upperCallback_);
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:26]: Rollback it please. It is against our code style

File: test/mock/mock_IGameCallback.h [hunk 2]
```diff
@@ -23,10 +23,10 @@ class GameCallbackMock : public IGameCallback
...
+	GameCallbackMock(UpperCallback* upperCallback_);
 	virtual ~GameCallbackMock();
 
-	void setGameState(CGameState * gameState);
+	void setGameState(CGameState* gameState);
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:29]:
```suggestion
	void setGameState(CGameState * gameState);
```

File: test/mock/mock_IGameCallback.h [hunk 3]
```diff
@@ -35,63 +35,63 @@ class GameCallbackMock : public IGameCallback
...
-	void giveHeroBonus(GiveBonus * bonus) override {};
-	void setMovePoints(SetMovePoints * smp) override {};
-	void setManaPoints(ObjectInstanceID hid, int val) override {};
-	void giveHero(ObjectInstanceID id, PlayerColor player) override {};
-	void changeObjPos(ObjectInstanceID objid, int3 newPos, ui8 flags) override {};
-	void heroExchange(ObjectInstanceID hero1, ObjectInstanceID hero2) override {}; //when two heroes meet on adventure map
-	void changeFogOfWar(int3 center, ui32 radius, PlayerColor player, bool hide) override {};
-	void changeFogOfWar(std::unordered_set<int3, ShashInt3> &tiles, PlayerColor player, bool hide) override {};
+	void setObjProperty(ObjectInstanceID objid, int prop, si64 val) override {}
+	void showInfoDialog(InfoWindow* iw) override {}
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:39]:
```suggestion
	void showInfoDialog(InfoWindow * iw) override {}
```

File: test/mock/mock_IGameCallback.h [hunk 4]
```diff
@@ -35,63 +35,63 @@ class GameCallbackMock : public IGameCallback
...
+	void showInfoDialog(const std::string& msg, PlayerColor player) override {}
+
+	void changeSpells(const CGHeroInstance* hero, bool give, const std::set<SpellID>& spells) override {}
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:42]:
```suggestion
	void changeSpells(const CGHeroInstance * hero, bool give, const std::set<SpellID> & spells) override {}
```

File: test/mock/mock_IGameCallback.h [hunk 5]
```diff
@@ -35,63 +35,63 @@ class GameCallbackMock : public IGameCallback
...
+	bool removeObject(const CGObjectInstance* obj) override { return false; }
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:43]:
```suggestion
	bool removeObject(const CGObjectInstance * obj) override { return false; }
```

File: test/mock/mock_IGameCallback.h [hunk 6]
```diff
@@ -35,63 +35,63 @@ class GameCallbackMock : public IGameCallback
...
+	void setOwner(const CGObjectInstance* objid, PlayerColor owner) override {}
```
Comment (nullkiller) [inline test/mock/mock_IGameCallback.h:44]: And so on

Comment (soarqin) [reply test/mock/mock_IGameCallback.h]: ok, force re-pushed, sorry for having reformatted using my own code-style

**Pull request merged**